src/App.tsx

```diff
@@ -15,35 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Empty = '',
+  Length = 'id',
+  Alphabetically = 'alphabetically',
+}
+
+type Query = {
+  reverse: boolean;
+};
+
+function getVisibleGoods(
+  goods: Array<string>,
+  sortField: SortType,
+  query: Query = { reverse: false },
+) {
+  const resGoods = [...goods];
+
+  if (sortField) {
+    resGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabetically: {
+          return good1.localeCompare(good2);
+        }
+
+        case SortType.Length: {
+          return good1.length - good2.length;
+        }
+      }
+    });
+  }
+
+  if (query.reverse) {
+    resGoods.reverse();
+  }
+
+  return resGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Empty);
+  const [reverse, setReverse] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
for boolean variables: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig